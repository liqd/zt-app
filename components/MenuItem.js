import React from 'react'
import { TouchableWithoutFeedback } from 'react-native'
import { ListItem } from '@rneui/themed'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import { COLORS } from '../theme/colors'
import { SIZES } from '../theme/fonts'
import { styles } from './MenuItem.styles'

export const MenuItem = (props) => {
  const {item: { isCancel, isFirst, isLast, isAllowed }} = props
  const getContainerStyle = () => {
    if (isCancel) {
      return styles.cancelButton
    } else if ( isFirst && isLast ) {
      return styles.listButtonOnly
    } else if ( isFirst ) {
      return styles.listButtonFirst
    } else if ( isLast ) {
      return styles.listButtonLast
    } else {
      return styles.listButton
    }
  }

  if (isAllowed) {
    return (
      <ListItem
        Component={TouchableWithoutFeedback}
        onPress={props.item.action}
        bottomDivider={!isCancel}
        containerStyle={getContainerStyle()}
      >
        {props.item.icon ?
          <IconSLI
            name={props.item.icon}
            color={COLORS.primary}
            size={SIZES.md}
          /> : null}
        <ListItem.Content style={isCancel ? styles.cancelContent : null}>
          <ListItem.Title
            style={isCancel ? styles.cancelButtonText : null}
          >
            {props.item.title}
          </ListItem.Title>
        </ListItem.Content>
      </ListItem>
    )
  } else {
    return null
  }
}
