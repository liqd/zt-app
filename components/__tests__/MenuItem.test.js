import React from 'react'
import { render } from '@testing-library/react-native'

import { MenuItem } from '../MenuItem'

test('Test MenuItem notAllowed', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: false, isLast: false, isAllowed: false}}
    />)
  expect(toJSON()).toBe(null)
})

// Snapshot tests
test('Test MenuItem isAllowed', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: false, isLast: false, isAllowed: true}}
    />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test MenuItem isAllowed with Icon', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: false, isLast: false, isAllowed: true, icon: 'check'}}
    />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test MenuItem isAllowed isCancel', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: true, isFirst: false, isLast: false, isAllowed: true}}
    />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test MenuItem isAllowed isFirst', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: true, isLast: false, isAllowed: true}}
    />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test MenuItem isAllowed isLast', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: false, isLast: true, isAllowed: true}}
    />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test MenuItem isAllowed isFirst and isLast', () => {
  const { toJSON} = render(
    <MenuItem
      item={{isCancel: false, isFirst: true, isLast: true, isAllowed: true}}
    />)
  expect(toJSON()).toMatchSnapshot()
})
