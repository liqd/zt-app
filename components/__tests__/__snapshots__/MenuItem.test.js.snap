// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MenuItem isAllowed 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderBottomWidth": 0.5,
      "borderColor": "#bcbbc1",
      "borderTopLeftRadius": 0,
      "borderTopRightRadius": 0,
      "flexDirection": "row",
      "padding": 14,
      "width": "90%",
    }
  }
  testID="padView"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        null,
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#242424",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test MenuItem isAllowed isCancel 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#bcbbc1",
      "flexDirection": "row",
      "padding": 14,
    }
  }
  testID="padView"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test MenuItem isAllowed isFirst 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderBottomWidth": 0.5,
      "borderColor": "#bcbbc1",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "row",
      "padding": 14,
      "width": "90%",
    }
  }
  testID="padView"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        null,
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#242424",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test MenuItem isAllowed isFirst and isLast 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderBottomWidth": 0.5,
      "borderColor": "#bcbbc1",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "row",
      "padding": 14,
      "width": "90%",
    }
  }
  testID="padView"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        null,
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#242424",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test MenuItem isAllowed isLast 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderBottomWidth": 0.5,
      "borderColor": "#bcbbc1",
      "flexDirection": "row",
      "padding": 14,
      "width": "90%",
    }
  }
  testID="padView"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        null,
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#242424",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test MenuItem isAllowed with Icon 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderBottomWidth": 0.5,
      "borderColor": "#bcbbc1",
      "borderTopLeftRadius": 0,
      "borderTopRightRadius": 0,
      "flexDirection": "row",
      "padding": 14,
      "width": "90%",
    }
  }
  testID="padView"
>
  <Text />
  <View
    style={
      Object {
        "paddingLeft": 16,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        },
        null,
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#242424",
          "fontSize": 17,
        }
      }
      testID="listItemTitle"
    >
      
    </Text>
  </View>
</View>
`;
